analyzers/src/SonarAnalyzer.Common/Helpers/ProjectTypeHelper.cs

```diff
@@ -29,11 +29,20 @@ internal static class ProjectTypeHelper
     {
         private static readonly ISet<string> TestAssemblyNames = new HashSet<string>
         {
+            "DOTMEMORY.UNIT",
             "MICROSOFT.VISUALSTUDIO.TESTPLATFORM.TESTFRAMEWORK",
             "MICROSOFT.VISUALSTUDIO.QUALITYTOOLS.UNITTESTFRAMEWORK",
             "XUNIT", // Legacy Xunit (v1.x)
             "XUNIT.CORE",
-            "NUNIT.FRAMEWORK"
+            "NUNIT.FRAMEWORK",
+            // Assertion
+            "FLUENTASSERTIONS",
+            "SHOULDLY",
+            // Mock
+            "FAKEITEASY",
```

Comment:
And there's more here: https://github.com/dariusz-wozniak/List-of-Testing-Tools-and-Frameworks-for-.NET/blob/master/README.md

Including web-testing, Azure functions testing, database testing etc.

It also mentions which have been discontinued.

Reply:
1M downloads is a threshold for me. To compare "NuGet VS VSIX" downloads for the most downloaded version:
[NUnitLite](https://www.nuget.org/stats/packages/NUnitLite?groupby=ClientName) - 1.9M total, for v3.12 it has  931 VSIX.
[NCrunch.Framework](https://www.nuget.org/stats/packages/NCrunch.Framework?groupby=Version&groupby=ClientName) - 730k total, for v4.5 it has 214 VSIX

As each referenced NuGet costs us a UT runtime for each build and very specific libraries could be used together with more common libraries for the rest of the project, I would not try to cover everything.

Reply:
These will be will accompanied by other tools we detect:
Faker.Net, NBuilder, Bogus - data generators
WireMock.Net - HTTP server mock
Sitecore.FakeDb - DB mock

Reply:
Ok, you can also double check with @agigleux .

> As each referenced NuGet costs us a UT runtime for each build
This is not an argument.

> very specific libraries could be used together with more common libraries for the rest of the project

This is relevant for any mocking library, right? So you should remove MOQ, RHINO.MOCKS and NSUBSTITUTE, to be consistent. If not, please add the others that match the general threshold, as well.

Reply:
Also, Machine Specifications (and possibly other BDD tools) don't seem to require a different test library... at least if I look at [this csproj](https://github.com/jonhilt/MSpec-Example/blob/master/src/MSpecExample.Tests/MSpecExample.Tests.csproj#L36) (am I missing something)?

Reply:
I'm about to add `Machine.Specifications`, `NUnitLite`, `SpecFlow`, `JustMock` as they match the purpose & usage criteria.
We simply cannot add all and there's always a way how to configure everything else manually, so let's use "The list isn't perfect when there's nothing to add, but when there's nothing to remove" principle.

Reply:
RE: `Sitecore.FakeDb`
- there could be a library extending its capabilities, and it may be considered as test code as well
- on the other hand, it might be used for "DEBUG" mode by applications, so probably it's better to not include these sort of libraries, indeed